modules/userId/index.js

```diff
@@ -309,13 +315,16 @@ function deleteValueFromLocalStorage(submodule) {
 }
 
 export function deleteStoredValue(submodule) {
-  const storageType = submodule.config?.storage?.type;
-
-  if (storageType === COOKIE) {
-    deleteValueFromCookie(submodule);
-  } else if (storageType === LOCAL_STORAGE) {
-    deleteValueFromLocalStorage(submodule);
-  }
+  calculateEnabledStorageTypes(submodule).forEach(storageType => {
```

Comment:
I'm wondering why methods like `setStoredValue` and `getStoredValue` can reference `submodule.enabledStorageTypes` whereas `deleteStoredValue` needs to call `calculateEnabledStorageTypes(...)`?

Reply:
that's a good question, `deleteStoredValue` is executed from `requestDataDeletion` and that function is exported.
specs are testing that function directly so we end up in an scenario where submodule.enabledStorageTypes never had the chance to be populated... enabledStorageTypes is populated from `initSubmodules` and the spec doesn't make the call to initSubmodules

don't know if in a real scenario `requestDataDeletion` could be called without going throw `initSubmodules` first, I tried to reproduce an scenario where `requestDataDeletion` is executed but without success.. from what I see in the code it does seem the function could be called without going through the `initSubmodules` call first, but I am not 100% sure.

One thing I could do is to do only call `calculateEnabledStorageTypes` if the `enabledStorageTypes` doesn't exist.

e.g.
```
 function deleteStoredValue(submodule) {
   const enabledStorageTypes = submodule.enabledStorageTypes || calculateEnabledStorageTypes(submodule)
  enabledStorageType.forEach(function (storageType) {
     switch (storageType) {
       case COOKIE:
         deleteValueFromCookie(submodule);
         break;
       case LOCAL_STORAGE:
         deleteValueFromLocalStorage(submodule);
         break;
     }
   });
 }
```

Reply:
Gotcha. I think that `getStoredValue`, `setStoredValues`, and `deleteStoredValue` should be consistent in their assumptions about whether `submodule.enabledStorageTypes` is populated -- i.e. either they should all use `submodule.enabledStorageTypes` or all use `calculateEnabledStorageTypes`. So you can either:
a. Use `calculateEnabledStorageTypes` in all three functions, OR
b. Populate `submodule.enabledStorageTypes` in `requestDataDeletion`, then use `submodule.enabledStorageTypes` in all three functions

Using `submodule.enabledStorageTypes || calculateEnabledStorageTypes(submodule)` in `deleteStoredValue` only still causes some confusion about why `submodule.enabledStorageTypes` might not be available there. You could duplicate this logic to all three functions, but at that point you might as well just call `calculateEnabledStorageTypes` in all three functions (option a).

Reply:
decided to go with option b.

I tried option a, but removing `submodule.enabledStorageTypes` entirely end up with many changes that make this a solution a bit more complicated, it's not only the call to `calculateEnabledStorageTypes(submodule)` on each get/set/delete fn.. it's also about changing the logic of `initSubmodules` function where there's one filter loop where `enabledStorageTypes` is calculated in order to use its value inside `canUseStorage()` and there's another loop that calls `populateSubmoduleId` where we'll need to have make another call to `calculateEnabledStorageTypes` in order to pass the types to the submodule getId function.